src/Controller/ConfigController.php

```diff
@@ -124,25 +128,27 @@ public function listAction(Request $request)
      * @param Request $request
      *
      * @return JsonResponse
+     *
+     * @throws \Exception
      */
-    public function deleteAction(Request $request)
+    public function deleteAction(Request $request): ?JsonResponse
     {
-        $this->checkPermission('plugin_datahub_config');
+        $this->checkPermission(self::CONFIG_NAME);
 
         try {
             $name = $request->get('name');
 
             $config = Dao::getByName($name);
-            if (empty($config)) {
-                throw new Exception('Name does not exist.');
+            if ($config instanceof Configuration) {
```

Comment:
really ?

Reply:
`getByName()` is documented as returning `@return Configuration|null` (I also checked manually) and I followed PHPStorm's inspection recommendation:

"'empty(...)' counts too many values as empty, consider refactoring with type sensitive checks."

Since `getByName()` should only return a matching object or null and null will never be an instance of anything the code works.

Reply:
I am referring to this change

            if ($config instanceof Configuration) {
                throw new \Exception('Name does not exist.');
            }

Reply:
I was unsure whether or not you wanted to use `\Exception` or `Symfony\Component\Config\Definition\Exception\Exception` but settled for the former because `Pimcore\Bundle\DataHubBundle\Configuration` throws `\Exception` and the docs in this file said `@throws \Exception` [here](https://github.com/pimcore/data-hub/blob/master/src/Controller/ConfigController.php#L244) and [here](https://github.com/pimcore/data-hub/blob/master/src/Controller/ConfigController.php#L277) and [here](https://github.com/pimcore/data-hub/blob/master/src/Controller/ConfigController.php#L367).

That is why I normalised the usage to \Exception for the Dao-related code.

Reply:
Still, maybe the ! is missing. If $config is instanceof Configuration then it exists. The exception claims it does not exist.

Reply:
Oh, right! 😄